src/test/java/teammates/test/cases/BaseTestCaseWithBackDoorApiAccess.java

```diff
@@ -10,26 +10,55 @@
 import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
 import teammates.common.datatransfer.attributes.InstructorAttributes;
 import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.util.Const;
 import teammates.test.driver.BackDoor;
+import teammates.test.driver.retry.RetryManager;
+import teammates.test.driver.retry.RetryableTaskReturns;
 
 /**
  * Base class for all test cases which are allowed to access the Datastore via {@link BackDoor}.
  */
 public abstract class BaseTestCaseWithBackDoorApiAccess extends BaseTestCaseWithDatastoreAccess {
 
+    protected AccountAttributes getAccount(String googleId) {
```

Comment:
I was thinking to push the trying logic to the abstract class `BaseTestCaseWithDatastoreAccess`. `BaseTestCaseWithDatastoreAccess` is functioned like an interface (It doesn't care whether you use `BackDoor` http request with a secret key or `BackDoor` logic to access datastore directly). I think there is also need for component tests to have retry (I think it is already there, in `BaseTestCaseWithDatastoreAccess#verifyPresentInDatastore()`).

BTW, is it necessary to write a new method `protected AccountAttributes getAccount(String googleId)`?